src/Microsoft.AspNetCore.Mvc.Core/ModelBinding/Binders/ComplexTypeModelBinder.cs

```diff
@@ -47,6 +48,27 @@ public Task BindModelAsync(ModelBindingContext bindingContext)
                 return TaskCache.CompletedTask;
             }
 
+            // The following check causes the ComplexTypeModelBinder to NOT participate in binding structs as 
+            // reflection does not provide information about the implicit parameterless constructor for a struct.
+            // This binder would eventually fail to construct an instance of the struct as the Linq's NewExpression
+            // compile fails to construct it.
+            var modelTypeInfo = bindingContext.ModelType.GetTypeInfo();
+            if (bindingContext.Model == null
+                && (modelTypeInfo.IsAbstract
+                || modelTypeInfo.GetConstructor(Type.EmptyTypes) == null))
+            {
+                if (bindingContext.IsTopLevelObject)
+                {
+                    throw new InvalidOperationException(
+                        Resources.FormatComplexTypeModelBinder_NoParameterlessConstructor_TopLevelObject(modelTypeInfo.FullName));
+                }
+
+                throw new InvalidOperationException(
+                    Resources.FormatComplexTypeModelBinder_NoParameterlessConstructor_ForProperty(
+                        modelTypeInfo.FullName,
+                        bindingContext.ModelName));
```

Comment:
This isn't going to spit out a correct error message. This code deals with
``` c#
public class Model
{
    public NoParameterlessConstructor Property { get; set; }
}
```
and the message should be something like
```
Could not create an instance of type 'NoParameterlessConstructor'. Model bound complex types must not be abstract or value types and must have a parameterless constructor. Alternatively, set the 'Property' property to a non-null value in the 'Model' constructor.
```